src/toolbox/AreaOfInterestDrawer.js

```diff
@@ -534,17 +523,21 @@ class NgmAreaOfInterestDrawer extends LitElementI18n {
         if (this.interestAreasDataSource.entities.contains(pickedObject.id)) {
           this.pickArea_(pickedObject.id.id);
         } else if (this.selectedArea_) {
-          this.updateHighlight(this.selectedArea_, false);
-          this.selectedArea_ = null;
+          this.deselectArea();
         }
       }
     }
   }
 
   deselectArea() {
-    if (this.selectedArea_ && !this.draw_.entityForEdit) {
+    if (this.selectedArea_) {
       this.updateHighlight(this.selectedArea_, false);
       this.selectedArea_ = null;
+      const objectInfo = document.querySelector('ngm-object-information');
```

Comment:
You should dispatch an event. The app component should catch it and do the work.
Could you please also fix "src/query/QueryManager.js"?